src/lib/mongoBackend/safeBsonGet.cpp

```diff
@@ -105,8 +102,31 @@ int getIntField(const BSONObj& b, const std::string& field)
 
 /* ****************************************************************************
 *
-* getBoolField -
+* getLongField -
+*/
+long getLongField(const BSONObj& b, const std::string& field)
```

Comment:
Being a littler slower (just internal function calls and ifs) is a very small price to pay for getting safer code. Remember the parachute metaphore ;)